pkg/storage/store.go

```diff
@@ -196,8 +197,22 @@ func (s *store) storeForPeriod(p config.PeriodConfig, chunkClient client.Client,
 		schema = index.NewSchemaCaching(schema, time.Duration(s.storeCfg.CacheLookupsOlderThan))
 	}
 
-	return series.NewWriter(f, s.schemaCfg, idx, schema, s.writeDedupeCache, s.storeCfg.DisableIndexDeduplication),
-		series.NewIndexStore(s.schemaCfg, schema, idx, f, s.cfg.MaxChunkBatchSize),
+	var (
+		writer       stores.ChunkWriter = series.NewWriter(f, s.schemaCfg, idx, schema, s.writeDedupeCache, s.storeCfg.DisableIndexDeduplication)
+		seriesdIndex *series.IndexStore = series.NewIndexStore(s.schemaCfg, schema, idx, f, s.cfg.MaxChunkBatchSize)
+		index        stores.Index       = seriesdIndex
+	)
+	if s.cfg.BoltDBShipperConfig.Mode == shipper.ModeReadOnly && s.cfg.BoltDBShipperConfig.IndexGatewayClientConfig.Address != "" {
+		// inject the index-gateway client into the index store
+		gw, err := shipper.NewGatewayClient(s.cfg.BoltDBShipperConfig.IndexGatewayClientConfig, indexClientReg)
```

Comment:
Wouldn't `idx` already be a gateway client instance created by `NewIndexClient` [here](https://github.com/JordanRushing/loki/blob/main/pkg/storage/factory.go#L148)?

Reply:
Yes but a different one. This one wrap the index-store, the other one call the old API.

Reply:
yeah, but that part is done in `IndexGatewayClientStore`, right?
you can type cast `idx` to `*shipper.GatewayClient` and pass it to `NewIndexGatewayClientStore` like below:
`series.NewIndexGatewayClientStore(idx.(*shipper.GatewayClient), seriesdIndex)`

Reply:
it might work but this is more risky.